great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3130,6 +3136,29 @@ def _determine_if_expectation_validation_result_include_rendered_content(
                 return False
         return include_rendered_content
 
+    @staticmethod
+    def _determine_save_changes_flag(save_changes: Optional[bool]) -> bool:
+        """
+        This method is meant to enable the gradual deprecation of the `save_changes` boolean
+        flag on various Datasource CRUD methods. Moving forward, we will always persist changes
+        made by these CRUD methods (a.k.a. the behavior created by save_changes=True).
+
+        As part of this effort, `save_changes` has been set to `None` as a default value
+        and will be automatically converted to `True` within this method. If a user passes in a boolean
+        value (thereby bypassing the default arg of `None`), a deprecation warning will be raised.
+        """
+        if save_changes is not None:
+            # deprecated-v0.15.32
+            warnings.warn(
+                'The parameter "save_changes" is deprecated as of v0.15.32; moving forward, '
+                "changes made to Datasources are always persisted by the underlying Store implementation. "
+                "This is part of an ongoing effort to standardized the behavior of our APIs. "
+                "As support will be removed in v0.18, please omit the argument moving forward.",
```

Comment:
What's the actual timeline on this deprecation? I don't think we can wait this long. cc @donaldheppner

Reply:
I think this message is overly long.
I would try to shorten it up and maybe link to a description of the changes somewhere, like a change-log or a slack post etc.